### PR TITLE
Add a section to the guide about map alignment with plotting.

### PR DESCRIPTION
**Pull request by Cadair**
This follows on from an issue that was raised on Twitter, about how the lack of alignment information breaks things.

### CODE REVIEW

File: doc/source/guide/data_types/maps.rst [hunk 1]
```diff
@@ -128,6 +128,14 @@ matplotlib figure object by using the plot() command.
 This makes it possible to use the SunPy plot as the foundation for a 
 more complicated figure.
 
+.. note::
+    
+   Currently the `~sunpy.map.GenericMap.plot()` and `~sunpy.map.GenericMap.peek()`
+   methods assume that the data is not rotated,
+   i.e. the solar y axis is aligned to the columns of the array. If this condition
+   is not met, when the map is plotted a warning will be issued. You can align 
+   the data by calling `~sunpy.map.GenericMap.rotate()` before you plot the Map.
```
Comment (dpshelio) [inline doc/source/guide/data_types/maps.rst]: do you have just to call it? or to update the map? 
I mean:
```python
map0.rotate()
```
or 
```python
map_rot = map0.rotate()
```

Comment (Cadair) [inline doc/source/guide/data_types/maps.rst]: the latter. I will re-word.

**Comment by Cadair** [comment]
@dpshelio updated

**Comment by ayshih** [comment]
I prefer the word "oriented" over the word "aligned", and the SunPy warning also uses "oriented" (because of me).  From my perspective, "aligned" invokes connotations of transverse positioning, which we don't do, and only weakly suggests rotation/roll.

I know it's semantics, but let's at least have the documentation use the same word as the code warning.

**Comment by Cadair** [comment]
@ayshih changed.

**Comment by ayshih** [comment]
:+1:

**Pull request merged**